Sources/AblyChat/DefaultMessages.swift

```diff
@@ -0,0 +1,265 @@
+import Ably
+
+
+public struct MessageEventPayloadWrapper: Hashable, Sendable {
+    let id: String
+    let listener: @Sendable (MessageEventPayload) -> Void
+
+    // Hashable conformance via the id
+    public func hash(into hasher: inout Hasher) {
+        hasher.combine(id)
+    }
+
+    public static func == (lhs: MessageEventPayloadWrapper, rhs: MessageEventPayloadWrapper) -> Bool {
+        return lhs.id == rhs.id
+    }
+    
+    public init(id: String, listener: @Sendable @escaping (MessageEventPayload) -> Void) {
+        self.id = id
+        self.listener = listener
+    }
+}
+
+typealias fromSerial = String
+typealias MessageListeners = [MessageEventPayloadWrapper: fromSerial]
+
+
+public struct MessageEventPayload: Sendable, Hashable {
+    
+    /**
+     * The type of the message event.
+     */
+    let type: MessageEvents
+
+    /**
+     * The message that was received.
+     */
+    public let message: Message
+}
+
+
+actor DefaultMessages: Messages, HandlesDiscontinuity {
```

Comment:
I don't know what most of the code in this file does; hopefully it will become clearer once there are spec references.